nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/providers/zookeeper/ZooKeeperStateProvider.java

```diff
@@ -167,11 +289,26 @@ synchronized ZooKeeper getZooKeeper() throws IOException {
         }
 
         if (zooKeeper == null) {
-            zooKeeper = new ZooKeeper(connectionString, timeoutMillis, new Watcher() {
-                @Override
-                public void process(WatchedEvent event) {
+            if(zooKeeperClientConfig != null && zooKeeperClientConfig.getClientSecure()) {
+                SecureClientZooKeeperFactory factory = new SecureClientZooKeeperFactory(zooKeeperClientConfig);
+                try {
+                    zooKeeper = factory.newZooKeeper(connectionString, timeoutMillis, new Watcher() {
+                        @Override
+                        public void process(WatchedEvent event) {
+                        }
+                    }, true);
+                } catch (Exception e) {
+                    System.out.println("Secure Zookeeper configuration failed!");
+                    e.printStackTrace();
```

Comment:
The exception should be sent to the logger instead of standard output and standard error pipes.